### PR TITLE
pr(archai): Changes the shape ordering to: batch_size x seq_len.

### PR DESCRIPTION
**Pull request by gugarosa**
This PR addresses the following issues:

* Changes the shape of input tensors to `[batch_size, seq_len]` instead of the opposite. Such improvement will allow for a more cleaner interface and allow users in using a shape standard that has been proposed by the NLP/Transformer-based community;
* Fixes the creation of `attention_masks` by including the length of `past_key_values` whenever they are used;
* Fixes `wt2` to `wt103` in configuration file. We should be using the same dataset identifier since the file explicitly says that is WikiText-103 for Transformer-XL Base;
* Removes unused key `num_attention_heads` from ONNX export.

*Note that to change the shape of input tensors, several files needed to be modified, including the ones from `scoring` and `nvidia_utils`* packages.

### CODE REVIEW

File: archai/nlp/nvidia_transformer_xl/mem_transformer.py [hunk 1]
```diff
@@ -811,7 +811,7 @@ def _forward(self, dec_inp, mems=None, past_key_values=None):
         plen = past_key_values[0][0].size(0) if past_key_values[0] is not None else 0
         klen = mlen + qlen
         if self.same_length:
-            all_ones = word_emb.new_ones(qlen, klen)
+            all_ones = word_emb.new_ones(qlen, klen+plen)
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/mem_transformer.py:814]: Could you please add comment on why this change was done?

File: archai/nlp/nvidia_transformer_xl/mem_transformer.py [hunk 2]
```diff
@@ -935,7 +941,7 @@ def forward(self, data:torch.Tensor, target:Optional[torch.Tensor], mems:Optiona
             loss = -F.log_softmax(logit, -1)[:, :, 0]
         else:
             loss, log_prob = self.crit(hidden=pred_hid.view(-1, pred_hid.size(-1)),
-                                       target=target.view(-1) if target is not None else None,
+                                       target=target.contiguous().view(-1) if target is not None else None,
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/mem_transformer.py:944]: Please add comment why contiguous is needed.

File: archai/nlp/nvidia_transformer_xl/wt103_base.yaml [hunk 3]
```diff
@@ -122,7 +122,7 @@ dgx1_1gpu_fp32: &dgx1_1gpu_fp32
 toy: &toy
    train:
       <<: *train
-      dataset: wt2
+      dataset: wt103
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/wt103_base.yaml:125]: This is only for toy mode. I can explain this in next meeting.

**Review by sytelus — COMMENTED** [review]
Overall looks great! I would suggest to add comments as noted on specific lines. Did you verify that default with WT103 re-produces baseline?

File: archai/nlp/nvidia_transformer_xl/wt103_base.yaml (hunk 3 above)
Comment (gugarosa) [reply archai/nlp/nvidia_transformer_xl/wt103_base.yaml]: I have removed the commit with such modification from the PR.

**Comment by gugarosa** [comment]
> Overall looks great! I would suggest to add comments as noted on specific lines. Did you verify that default with WT103 re-produces baseline?

Yes, definitely! Both previous and updated versions achieves the same metric values.

File: archai/nlp/nvidia_transformer_xl/mem_transformer.py (hunk 2 above)
Comment (gugarosa) [reply archai/nlp/nvidia_transformer_xl/mem_transformer.py:948]: I have added a new commit with the following comment on why contiguous is needed:

"As we are transposing the `target`, we need it in a contiguous piece of memory before applying a tensor visualization (view)"

File: archai/nlp/nvidia_transformer_xl/mem_transformer.py (hunk 1 above)
Comment (gugarosa) [reply archai/nlp/nvidia_transformer_xl/mem_transformer.py:816]: I have added a new commit with the following comment on why we need to add `plen`:

"`plen` should be taken into account when creating the attention mask because `past_key_values` might be used."
